generators/utils.js

```diff
@@ -606,3 +607,25 @@ function normalizeBlueprintName(blueprint) {
     }
     return blueprint;
 }
+
+/**
+ * Calculate a hash code for a given string.
+ * @param {string} str - any string
+ * @returns {number} returns the calculated hash code.
+ */
+function stringHashCode(str) {
```

Comment:
this is hard to understand

Reply:
I see this has been taken from https://stackoverflow.com/questions/7616461/generate-a-hash-from-string-in-javascript
Copy/pasting is actually bad in this case:
  - why do we need to declare all the variables outside the loop?
  - the naming has to be improved (chr => character, for instance)
  - there should be a comment on where this snippet has been taken from and adapted, or at least to say this has been adapted from what exists in Java
  - what about `Math.imul` ([resource](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/imul))

Reply:
I took it from here:
https://github.com/jhipster/generator-jhipster/blob/e362242b0f750f04867fb2e087f1911af84e548b/generators/generator-base.js#L1673-L1686

Reply:
Extract as an util function

Reply:
fair enough

Reply:
can you at least improve the code?

Reply:
done, I should reference stackoverflow on this?

Reply:
Maybe not

Reply:
Is this resolved?